src/SAWScript/Crucible/MIR/Builtins.hs

```diff
@@ -1089,18 +1054,156 @@ setupCrucibleContext rm =
+               ]
+     Refl <- case testEquality staticInitRetTy Crucible.UnitRepr of
+       Just e ->
+         pure e
+       Nothing ->
+         panic "setupCrucibleContext"
+               [ "static initializer should return ():"
+               , show staticInitRetTy
+               ]
+
+     let bindings = Crucible.fnBindingsFromList $
+                    map (\(Crucible.AnyCFG cfg) ->
+                          Crucible.FnBinding
+                            (Crucible.cfgHandle cfg)
+                            (Crucible.UseCFG cfg (Crucible.postdomInfo cfg))) $
+                    Map.elems cfgMap
+     let simctx0 = Crucible.initSimContext bak
+                     Mir.mirIntrinsicTypes halloc stdout
+                     bindings Mir.mirExtImpl
+                     SAWCruciblePersonality
+     let globals0 = Crucible.emptyGlobals
+     let setupGlobals = Crucible.regValue <$> Crucible.callCFG staticInitCfg Crucible.emptyRegMap
+     -- Step (1) in Note [Translating MIR statics in SAW]
+     (simctx1, gp) <- io $ runCrucible opts simctx0 globals0 [] Crucible.UnitRepr setupGlobals
+     let globalsAllStatics = gp ^. Crucible.gpGlobals
+     -- Step (2) in Note [Translating MIR statics in SAW]
+     let (globalsImmutStaticsOnly, staticInitializerPairs) =
+           mapAccumL
+             (\globals (staticDefId, Mir.StaticVar gv) ->
+               let static =
+                     case Map.lookup staticDefId (col ^. Mir.statics) of
+                       Just static' ->
+                         static'
+                       Nothing ->
+                         panic "setupCrucibleContext"
+                               [ "staticDefId not in statics map:"
+                               , show staticDefId
+                               ] in
+               case Crucible.lookupGlobal gv globalsAllStatics of
+                 Just rv ->
+                   let pair = MapF.Pair gv (Crucible.RV rv) in
+                   if static ^. Mir.sMutable
+                     then (globals, pair)
+                     else (Crucible.insertGlobal gv rv globals, pair)
+                 Nothing ->
+                   panic "setupCrucibleContext"
+                         [ "Static GlobalVar not in SymGlobalState:"
+                         , show gv
+                         ])
+             globals0
+             (cs ^. Mir.staticMap . to Map.toList)
+     let staticInitializerMap = MapF.fromList staticInitializerPairs
+
      return MIRCrucibleContext { _mccRustModule = rm
-                               , _mccBackend = bak
-                               , _mccHandleAllocator = halloc
+                               , _mccBackend = someBak
+                               , _mccSimContext = simctx1
+                               , _mccSymGlobalState = globalsImmutStaticsOnly
+                               , _mccStaticInitializerMap = staticInitializerMap
                                }
 
+{-
+Note [Translating MIR statics in SAW]
```

Comment:
Translating static items is a somewhat involved process, so I wrote up a high-level description of this works in this Note.